app/src/main/res/layout/enrollment_activity.xml

```diff
@@ -18,259 +18,79 @@
-                android:id="@+id/fieldRecycler"
-                android:layout_width="match_parent"
-                android:layout_height="0dp"
-                android:clipToPadding="false"
-                android:paddingBottom="100dp"
-                app:layoutManager="androidx.recyclerview.widget.LinearLayoutManager"
-                app:layout_constraintBottom_toBottomOf="parent"
-                app:layout_constraintTop_toBottomOf="@id/enrollmentDataButton"
-                tools:listitem="@layout/form_unsupported" />
+                app:layout_constraintStart_toEndOf="@id/menu"
+                tools:text="TEI NAME"
+                tools:visibility="gone" />
 
             <include
-                android:id="@+id/cl_indicator_progress"
-                layout="@layout/indicator_progress"
-                android:layout_width="match_parent"
+                android:id="@+id/teiDataHeader"
+                layout="@layout/tei_data_header"
+                android:layout_width="0dp"
                 android:layout_height="wrap_content"
-                android:layout_marginStart="0dp"
-                android:visibility="gone"
-                app:layout_constraintTop_toBottomOf="@id/enrollmentDataButton"
-                tools:visibility="visible" />
+                app:layout_constraintBottom_toBottomOf="parent"
+                app:layout_constraintEnd_toEndOf="parent"
+                app:layout_constraintStart_toEndOf="@id/menu"
+                app:layout_constraintTop_toTopOf="parent"/>
 
         </androidx.constraintlayout.widget.ConstraintLayout>
 
+        <androidx.recyclerview.widget.RecyclerView
+            android:id="@+id/fieldRecycler"
+            android:layout_width="match_parent"
+            android:layout_height="0dp"
+            android:clipToPadding="false"
+            android:paddingBottom="100dp"
+            app:layoutManager="androidx.recyclerview.widget.LinearLayoutManager"
+            app:layout_constraintBottom_toBottomOf="parent"
+            app:layout_constraintTop_toBottomOf="@id/toolbar"
+            tools:listitem="@layout/form_unsupported" />
+
         <com.google.android.material.floatingactionbutton.FloatingActionButton
             android:id="@+id/save"
             style="@style/Widget.MaterialComponents.FloatingActionButton"
             android:layout_width="wrap_content"
             android:layout_height="wrap_content"
             android:layout_margin="16dp"
-            android:layout_gravity="end|bottom"
             app:backgroundTint="?colorPrimary"
+            app:layout_constraintBottom_toBottomOf="parent"
+            app:layout_constraintEnd_toEndOf="parent"
             app:srcCompat="@drawable/ic_save_alt"
-            app:tint="@color/white"
-            />
-    </androidx.coordinatorlayout.widget.CoordinatorLayout>
+            app:tint="@color/white" />
+
+    </androidx.constraintlayout.widget.ConstraintLayout>
```

Comment:
For the FAB to have animations for free it needs to be a child of a coordinator view or are we not having animations?

Reply:
The recyclerview was scrolling wierdly because of this